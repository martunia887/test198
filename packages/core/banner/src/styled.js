// @flow
import styled, { css } from 'styled-components';
import { themed, math } from '@atlaskit/theme';
import { R400, R300, Y300, N500, N0, DN40, N700 } from '@atlaskit/theme/colors';
import { gridSize } from '@atlaskit/theme/constants';

const TRANSITION_DURATION = '0.25s ease-in-out';

/* Container */
export const getMaxHeight = ({
  appearance,
}: {
  appearance: 'warning' | 'error' | 'announcement',
}) => (appearance === 'announcement' ? '88px' : '52px');

export const backgroundColor = themed('appearance', {
  error: { light: R400, dark: R300 },
  warning: { light: Y300, dark: Y300 },
  announcement: { light: N500, dark: N500 },
});

export const Container = styled.div`
  max-height: ${getMaxHeight};
  overflow: ${({ appearance }) =>
    appearance === 'announcement' ? 'scroll' : 'visible'};
  background-color: ${backgroundColor};
`;

export const testErrorBackgroundColor = R400;
export const testErrorTextColor = N0;

export const textColor = themed('appearance', {
  error: { light: N0, dark: DN40 },
  warning: { light: N700, dark: DN40 },
  announcement: { light: N0, dark: N0 },
});
export const Content = styled.div`
  align-items: center;
  background-color: ${backgroundColor};
  color: ${textColor};
  display: flex;
  fill: ${backgroundColor};
  font-weight: 500;
  justify-content: center;
  padding: ${math.multiply(gridSize, 2)}px;
  text-align: center;
  ${'' /* transition: color ${TRANSITION_DURATION}; */}

  margin: auto;
  ${({ appearance }) =>
    appearance === 'announcement'
      ? 'max-width: 876px;'
      : ''} transition: color ${TRANSITION_DURATION};

  a,
  a:visited,
  a:hover,
  a:active,
  a:focus {
    color: ${textColor};
    text-decoration: underline;
  }
`;

export const Icon = styled.span`
  align-items: center;
  display: flex;
  flex-direction: column;
  flex: 0 0 auto;
`;

const textOverflow = ({
  appearance,
}: {
  appearance: 'announcement' | 'warning' | 'error',
}) =>
  appearance === 'announcement'
    ? ''
    : css`
        text-overflow: ellipsis;
        white-space: nowrap;
      `;

export const Visibility = styled.div`
  max-height: ${({ bannerHeight, isOpen }) => (isOpen ? bannerHeight : 0)}px;
  overflow: hidden;
  transition: max-height ${TRANSITION_DURATION};
`;

export const Text = styled.span`
  flex: 0 1 auto;
  padding-left: ${math.divide(gridSize, 2)}px;
  ${textOverflow};
  overflow: hidden;
`;
