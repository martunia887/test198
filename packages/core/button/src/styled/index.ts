export { default as StyledButton } from './StyledButton';
export { default as ButtonWrapper } from './ButtonWrapper';
export { default as ButtonContent } from './ButtonContent';
export { default as LoadingSpinner } from './LoadingSpinner';
export { default as IconWrapper } from './IconWrapper';
