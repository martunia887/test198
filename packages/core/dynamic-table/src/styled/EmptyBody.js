// @flow
import styled from '@emotion/styled';
import { gridSize, math } from '@atlaskit/theme';

export const EmptyViewWithFixedHeight = styled.div`
  height: ${math.multiply(gridSize, 18)}px;
`;

export const EmptyViewContainer = styled.div`
  margin: auto;
  padding: 10px;
  text-align: center;
  width: 50%;
`;
