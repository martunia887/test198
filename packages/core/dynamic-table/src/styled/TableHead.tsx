import styled, { css } from 'styled-components';
import { B100 } from '@atlaskit/theme/colors';

import { head } from '../theme';
import { SortOrderType } from '../types';

import {
  arrowsStyle,
  cellStyle,
  onClickStyle,
  truncateStyle,
  TruncateStyleProps,
} from './constants';

const rankingStyles = css`
  display: block;
`;

interface HeadProps {
  isRanking?: boolean;
}

export const Head = styled.thead<HeadProps>`
  border-bottom: 2px solid ${head.borderColor};
  ${({ isRanking }) => isRanking && rankingStyles};
`;

interface HeadCellProps extends TruncateStyleProps {
  onClick?: () => void;
  isSortable?: boolean;
  sortOrder?: SortOrderType;
}

export const HeadCell = styled.th<HeadCellProps>`
  ${({ onClick }) => onClickStyle({ onClick: Boolean(onClick) })} 
  ${p => truncateStyle(p)} 
  ${p => arrowsStyle(p)} 
  ${cellStyle} 
  border: none;
  color: ${head.textColor};
  box-sizing: border-box;
  font-size: 12px;
  font-weight: 600;
  position: relative;
  text-align: left;
  vertical-align: top;
  &:focus {
    outline: solid 2px ${B100};
  }
`;
