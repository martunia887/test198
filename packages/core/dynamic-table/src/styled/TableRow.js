// @flow
import styled from '@emotion/styled';
import { row } from '../theme';

export const TableBodyRow = styled.tr`
  &:hover {
    background: ${row.hoverBackground};
  }
`;
