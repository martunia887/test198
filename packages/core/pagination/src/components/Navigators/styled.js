//@flow
import styled from '@emotion/styled';
import Button from '@atlaskit/button';
import { gridSize } from '@atlaskit/theme';

export const StyledButton = styled(Button)`
  [dir='rtl'] & {
    transform: rotate(180deg);
  }
  padding-left: ${gridSize() / 2}px;
  padding-right: ${gridSize() / 2}px;
`;
