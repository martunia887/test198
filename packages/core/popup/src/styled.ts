import { FC } from 'react';
import styled from '@emotion/styled';
import { borderRadius, layers } from '@atlaskit/theme/constants';
import { N0, N40A } from '@atlaskit/theme/colors';
import { WrapperContainerProps } from './types';
const shadow = N40A;

export const StyledPopup: FC<WrapperContainerProps> = styled.div`
  background-color: ${N0};
  border-radius: ${borderRadius()}px;
  box-shadow: 0 0 0 1px ${shadow}, 0 4px 11px ${shadow};
  z-index: ${layers.layer()};
  box-sizing: border-box;
  overflow: auto;
  display: block;
  :focus {
    outline: none;
  }
  flex: 1 1 auto;
`;

export const PopupRelContainer = styled.div`
  position: relative;
`;
