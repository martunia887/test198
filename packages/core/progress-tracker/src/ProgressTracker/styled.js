// @flow

import styled from '@emotion/styled';

export const ProgressTrackerContainer = styled.div`
  display: flex;
  width: 100%;
  margin-top: 40px;
`;

export default ProgressTrackerContainer;
