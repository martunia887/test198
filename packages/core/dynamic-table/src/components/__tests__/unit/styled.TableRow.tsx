import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { TableBodyRow } from '../../../styled/TableRow';

describe('TableRow', () => {
  let shallowWrapper: ShallowWrapper;

  describe('on rendering with highlighted prop', () => {
    beforeEach(() => {
      shallowWrapper = shallow(<TableBodyRow isHighlighted={true} />);
    });

    it('should render with required background color', () => {
      expect(shallowWrapper).toMatchSnapshot();
    });
  });

  describe('on rendering without highlighted prop', () => {
    beforeEach(() => {
      shallowWrapper = shallow(<TableBodyRow isHighlighted={false} />);
    });

    it('should render without background color', () => {
      expect(shallowWrapper).toMatchSnapshot();
    });
  });
});
