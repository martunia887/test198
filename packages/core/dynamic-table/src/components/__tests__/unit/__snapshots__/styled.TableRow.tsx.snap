// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableRow on rendering with highlighted prop should render with required background color 1`] = `
.c0 {
  background-color: #DEEBFF;
}

.c0:hover {
  background: #FAFBFC;
}

<tr
  className="c0"
/>
`;

exports[`TableRow on rendering without highlighted prop should render without background color 1`] = `
.c0:hover {
  background: #FAFBFC;
}

<tr
  className="c0"
/>
`;
